src/main/java/bio/terra/app/usermetrics/BardClient.java

```diff
@@ -23,30 +29,46 @@ public class BardClient {
 
   private final UserMetricsConfiguration metricsConfig;
 
-  private final RestTemplate restTemplate;
+  private final RestTemplate apiRestTemplate;
+
+  private final RestTemplate syncRestTemplate;
   private final HttpHeaders headers;
 
+  private static final int BEARER_TOKEN_CACHE_TIMEOUT_SECONDS = 3600;
+
+  private final Map<String, Object> bearerCache;
+
   @Autowired
   public BardClient(UserMetricsConfiguration metricsConfig) {
-    this.restTemplate = new RestTemplate();
-    restTemplate.setRequestFactory(new HttpComponentsClientHttpRequestFactory());
+    this.apiRestTemplate = new RestTemplate();
+    apiRestTemplate.setRequestFactory(new HttpComponentsClientHttpRequestFactory());
+    this.syncRestTemplate = new RestTemplate();
+    syncRestTemplate.setRequestFactory(new HttpComponentsClientHttpRequestFactory());
+
     this.headers = new HttpHeaders();
     headers.setContentType(MediaType.APPLICATION_JSON);
     headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON, MediaType.APPLICATION_JSON));
 
+    int ttl =
+        Objects.requireNonNullElse(
+            metricsConfig.getSyncRefreshInterval(), BEARER_TOKEN_CACHE_TIMEOUT_SECONDS);
+    this.bearerCache = Collections.synchronizedMap(new PassiveExpiringMap<>(ttl, TimeUnit.SECONDS));
```

Comment:
I wonder if it's OK to use `PassiveExpiringMap` in this way, as it's not thread safe and `BardClient` is a singleton. Is it only used on a single thread? If synchronization is needed I think wrapping in `SynchronizedMap` should provide that.

Reply:
I believe the BardClient is used by multiple threads via the metricsPerformanceThreadpool in the UserMetricsInterceptor.
It looks like the developers at Apache Commons support wrapping in the synchronized map . They note: 

> If you wish to use this map from multiple threads concurrently, you must use appropriate synchronization. The simplest approach is to wrap this map using [Collections.synchronizedMap(Map)](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html?is-external=true#synchronizedMap-java.util.Map-)